chirp/drivers/retevis_ra25.py

```diff
@@ -0,0 +1,1058 @@
+        rs = RadioSettingValueString(
+            minlength=4, maxlength=4, current=_current)
+        rs.set_mutable(False)
+        rset = RadioSetting("radio_info.ver", "Version", rs)
+        info_menu.append(rset)
+
+        _current = "".join(str(_immut_1.serial_number[i]) for i in range(
+            len(_immut_1.serial_number)))
+        rs = RadioSettingValueString(
+            minlength=0, maxlength=16, current=_current)
+        rs.set_mutable(False)
+        rset = RadioSetting("immut_1.serial_number", "Serial Number", rs)
+        info_menu.append(rset)
+
+        _current = "".join(str(_immut_2.prod_date[i]) for i in range(
+            len(_immut_2.prod_date)) if int(_immut_2.prod_date[i]) >= 0x20)
+        rs = RadioSettingValueString(
+            minlength=0, maxlength=10, current=_current)
+        rs.set_mutable(False)
+        rset = RadioSetting("immut_2.prod_date", "Production Date", rs)
+        info_menu.append(rset)
+        return group
+
+    def set_settings(self, settings):
+        for element in settings:
+            if not isinstance(element, RadioSetting):
+                self.set_settings(element)
+                continue
+            else:
+                try:
+                    if "." in element.get_name():
+                        toks = element.get_name().split(".")
+                        obj = self._memobj
+                        for tok in toks[:-1]:
+                            if '[' in tok:
+                                t, i = tok.split("[")
+                                i = int(i[:-1])
+                                obj = getattr(obj, t)[i]
+                            else:
+                                obj = getattr(obj, tok)
+                        setting = toks[-1]
+                    else:
+                        obj = self._memobj.settings
+                        setting = element.get_name()
+
+                    if element.has_apply_callback():
+                        log.debug("applying callback")
+                        element.run_apply_callback()
+
+                    if element.value.get_mutable():
+                        if obj._name in ['dtmf_list', 'dtmf_start', 'dtmf_end',
+                                         'remote_stun', 'remote_kill']:
+                            _val = [0] * len(obj[setting])
+                            _i = 0  # offset _mem setting idx for pad chars
+                            for i in range(len(element.value)):
+                                if element.value[i] == '*':
+                                    _val[i - _i] = ord('E')
+                                elif element.value[i] == '#':
+                                    _val[i - _i] = ord('F')
+                                elif element.value[i] in self.VALID_DTMF:
+                                    _val[i - _i] = ord(element.value[i])
+                                else:
+                                    _i += 1
```

Comment:
Seems better to encapsulate this in a helper like you did for the get case above, but... this is what I was asking for - do the proper conversion during set and keep the memory consistent with what is uploaded. Thanks for changing it.